packages/frontend-shared/src/composables/useCollapsibleTree.ts

```diff
@@ -38,20 +45,37 @@ function collectRoots<T extends RawNode<T>> (node: UseCollapsibleTreeNode<T> | n
 }
 
 export const useCollapsibleTreeNode = <T extends RawNode<T>>(rawNode: T, options: UseCollapsibleTreeOptions, depth: number, parent: UseCollapsibleTreeNode<T> | null): UseCollapsibleTreeNode<T> => {
+  const { cache, expandInitially } = options
   const treeNode = rawNode as UseCollapsibleTreeNode<T>
   const roots = parent ? collectRoots<T>(parent) : []
-  const [expanded, toggle] = useToggle(!!options.expandInitially)
+  const [expanded, toggle] = useToggle(cache?.get(rawNode.id) ?? !!expandInitially)
+
   const hidden = computed(() => {
     return !!roots.find((r) => r.expanded.value === false)
   })
 
+  let wrappedToggle = toggle
```

Comment:
Rather than conditionally reassigning a function, what about just a single function that has a conditional? This seems more idiomatic in general, and a bit easier to understand.

Reply:
@mike-plummer thoughts on this one?

Reply:
I think that makes sense - we aren't necessarily memoizing on the toggle function anywhere so there isn't a strong argument to avoid wrapping it. May as well always do the wrap and move the conditional inside